canonical-web-and-design/maas-ui
# Display action menu in machine list.

## Done
- Built non-interactive action menu in machine list. It displays a list of actions that the selected machines can perform along with a count of how many machines can perform it. If the count is 0 the action is filtered, unless it's a lifecycle action in which case it will show up disabled.
- Moved the `selectedMachines` state (and tests) up from `MachineList` to `Machines` so it can be shared with `HeaderStrip`. I'm not sure if it would be better for it to live in Redux. @squidsoup @huwshimi any thoughts?
- Removed `Routes` component so routing is now handled by `Machines`.

## QA
- Select a bunch of machines and click the take action menu.
- Check that the actions and counts match what's found in the angular version.

## Fixes
Fixes canonical-web-and-design/MAAS-squad#1778
Fixes canonical-web-and-design/MAAS-squad#1792

## Screenshots
### Disabled
![0 0 0 0_8400_MAAS_r_machines (11)](https://user-images.githubusercontent.com/25733845/74675172-4b260480-51b3-11ea-8142-1216592b8703.png)

### Machines selected
![0 0 0 0_8400_MAAS_r_machines (12)](https://user-images.githubusercontent.com/25733845/74675181-4eb98b80-51b3-11ea-8e23-fc63e4193eb6.png)

### Opened
![0 0 0 0_8400_MAAS_r_machines (13)](https://user-images.githubusercontent.com/25733845/74675187-50834f00-51b3-11ea-8c0d-60285f17bcba.png)
